crates/server/src/admin/handlers/proposals.rs

```diff
@@ -105,115 +100,147 @@ pub struct Message {
 #[derive(Debug, Deserialize, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct GetProposalsResponse {
-    pub data: Vec<Proposal>,
+    pub data: Vec<ProposalConfig>,
 }
 
 #[derive(Debug, Deserialize, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct GetProposalResponse {
-    pub data: Proposal,
+    pub data: ProposalConfig,
+}
+
+#[derive(Copy, Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetProposalsRequest {
+    pub offset: u32,
+    pub limit: u32,
 }
 
 pub async fn get_proposals_handler(
     Path(context_id): Path<String>,
-    session: Session,
     Extension(state): Extension<Arc<AdminState>>,
+    Json(req): Json<GetProposalsRequest>,
 ) -> impl IntoResponse {
-    let sample_action = Action::ExternalFunctionCall(ExternalFunctionCall {
-        receiver_id: get_mock_user(),
-        method_name: "sampleMethod".to_owned(),
-        args: serde_json::json!({"example": "value"}),
-        deposit: "100".to_owned(),
-        gas: "10".to_owned(),
-    });
-
-    let proposals = vec![Proposal {
-        id: "proposal_1".to_owned(),
-        author: get_mock_user(),
-        actions: vec![sample_action],
-        title: "Proposal 1".to_owned(),
-        description: "This is the first proposal.".to_owned(),
-        created_at: "2024-10-31T12:00:00Z".to_owned(),
-    }];
-
-    ApiResponse {
-        payload: GetProposalsResponse { data: proposals },
+    let context_id: ContextId = context_id.parse().expect("Invalid context_id format");
+
+    match state
+        .ctx_manager
+        .get_proposals(context_id, req.offset as usize, req.limit as usize)
```

Comment:
why not make these `usize` ahead of time?

Reply:
updated